### ISSUE TITLE
Pending Issues for First SWELL Release of Atmos Analysis

### ISSUE THREAD

**Issue by rtodling** [GEOS-ESM/swell#443]
This is a revised version of the original Epic Issue #236. The closed PRs associated with the original issue have been removed from the list below - at the time this revised version is being opened. 

- https://github.com/GEOS-ESM/swell/issues/241 (**All**)
- https://github.com/GEOS-ESM/swell/issues/246 (**Ricardo**)
- https://github.com/GEOS-ESM/swell/pull/422 (**Ricardo**)
- https://github.com/GEOS-ESM/swell/pull/312 (**Ricardo**, **Doruk**)
- https://github.com/GEOS-ESM/swell/issues/371 (**Ricardo**)
- https://github.com/JCSDA-internal/fv3-jedi/pull/1294 (**Ricardo**)
-  https://github.com/GEOS-ESM/swell/issues/239 (**Ricardo**, **Doruk**)
- https://github.com/GEOS-ESM/swell/issues/451 (**Ricardo**, **Min-Jeong**)
- #458 
- #461 (**Ricardo**,**Doruk**)
- #463 
- #464  Consistent calculation of tropopause pressure w/ that of GSI (**Ricardo**)
- #466 
- #467 
- PR #468 
- PR #471 
- #472 
- #475 
- #476 
- #431

List of things for which there are yet no explicit Issues or PRs on GitHub (since not been in GitHub I am explicitly putting down the name of :

- [ ] Implement Q2 options for moisture analysis variable.
   This issue relatively complex. From the perspective of GSIBEC there isn't anything to do; other than a reset of some defaults in the namelist used to configure GSIBEC. However, in order for those knobs, viz, qoption=2 and pseudo_q2, to work we need anything depending on the derivatives to work. (**Ricardo**)

- [x] The above requires a revision of the handling of the halo in the interface between GSIBEC and SABER. The easiest way to address the halo issue is to convert the present unstructured interpolation to a gridded-type interpolation option from Atlas.(**Ricardo**) - see also: https://github.com/JCSDA-internal/saber/issues/962

- [ ] implement TLNMC: though GSIBEC needs to be changed a little bit, the main hold up on this is the issue of the halo and the derivatives (tendencies) laid out in the previous item. (**Ricardo**)

- [ ] We need to understand why the observation counts for some of the radiance (particularly IR) is still considerably different from the numbers in GSI, but not in HofX/GeoVals mode; this needs to be understood and worked when the GeoVals seem by GSI are those from the underlying background. (**Wei**, **Yanqiu**, **Ricardo**)

- [x] Engage in the ongoing discussion of how to finalize implementation, trigger, use and cycled the aircraft bias correction. This is being discussed here: https://github.com/JCSDA-internal/ufo/pull/3481  (**Yanqiu**, **Meta**, **Ricardo**) (Meta's PR for this: https://github.com/GEOS-ESM/swell/pull/478/files)

- [ ] Fully exercise the cycling VarBC capability that's been made available in SWELL, finding ways to provide convincing evidence that VarBC is working as it should. Is the VarBC preconditioning working for us? (**Yanqiu**)

- [ ] Figure out why the assimilation of radiance is producing increments so largely different than those from GSI, especially in the stratosphere for MW and more pronouncedly so for IR (see also VarBC item below): https://github.com/GEOS-ESM/swell/issues/426  (**Wei**, **Yanqiu**, **Ricardo**)

- [ ] Add assimilation of Tropical Cyclone information - TC-Vitals - pls notice that the desired implementation of this is as follows: (i) add program to IODA converters to convert TCVITAL files to IODA; (ii) add yaml entries to handle such observation either as independent ob yaml - must be so since TCVITAL will be in a IODA file of its own; (iii) implement necessary adjustment of ob-error as filter to UFO, in a similar manner done in GSI; (iv) perform analysis with TCVITAL-only (single or multiple entries) and compare directly with same from GSI. (**Wei**, **Meta**, **Yanqiu**)

- [ ] Replace CRTM static files in GSI with something more along the lines of what JEDI is using - 20 surf types. https://github.com/GEOS-ESM/GEOSana_GridComp/issues/177  (**Ricardo**, **Min-Jeong**)

- [ ] Cubed output analysis files from JEDI cannot be fed back to JEDI as pretend-backgrounds. This is because there are netcdf attributes missing from the cubed analysis output files. **Ricardo** has a branch in fv3-jedi with the modifications needed to correct for this - need to file a PR.

- [ ] Add MLS-Temperature observations: this are data typically used passively in x-type experiments, but once passive implementation is done, it should also be simply to add knob for eventual assimilation of the data. Steps: (i) IODA converters, remember GMAO uses the neear-real-time product directly in netcdf form, therefore converter will need to be develop to go from netcdf to IODA with passing through the bufr layer; UFO QC and Obs-Error settings. (**Kris**, **Yanqiu**)

- [ ] Ability to compare two SWELL experiments - side-by-side plots with EVA - we should talk about types of plots, but for starters it would be nice to have plots comparing the existing plots that EVA produces (**Akira**); for example: 
   (i) analysis increment figures (would be a plot w/ three panels exp (top) control (middle), difference (top-middle) in the bottom. 
   (ii) time series of OmB's: Mean and RMS for exp and control
   (iii) maps of OmB and OmA, mean and RMS, for various instruments in the 3-panel format above: exp(top), control(middle), diff (bottom)

- [ ] For sometime we'll need to have the ability to test with both yaml settings from IODA files that have been created from GSI output (ncdiags), and simultaneously test with IODA files that have been created from the bufr converters; work that Seth is doing. The issue here is to find a way to accommodate testing the atmospheric analysis options running from either one of these, and to find a way to manage the yaml files - ideally the handling of bufr input would not affect the settings already existing in the obs-yamls we presently have in swell; however, something tells me this is not going to be the case, i.e., the handling of the bufr-IODA option might actually require a revision to some of the existing filters. In such cases, an alternative location for new (duplicate) yaml files will need to be handled in SWELL.  (**Doruk**, **Seth**, **Ricardo**)

- [ ] Similarly to the above, there will be need to keep a set of ops-yaml files that are configured for the purposes of OSSEs (**Min-Jeong**), reanalysis (**Amal**, **Erin**), and others, and thus fall in the same category of having SWELL manage different options for the observing system settings. We could refer to these various configurations of the observing system as Arlindo refers to them, that is, as: scenarios. 

- [ ] Need to straighten up minor discrepancy between CRTM-v2.4.1-jedi-1 used in GEOS-GSI (v5.41.x) and GMAO-JEDI: there are the clouds coefficients from MJK and the spectral coefficients for amsua-metop-c which does not have what it takes to run both Ta and Tb cases. (**Ricardo**, **Doruk**)

- [ ] Need to add handles to get 3dvar and 3dfgat - atmos - to have Skin Temperature in the analysis vector. (**Ricardo**)

**Comment by mjkagnes123** [comment]
While comparing x0048 and SWELL analysis increments generated by assimilating each satellite sensor, the GPSRO increments looked too different, with opposite signs, at least for the Ps increment.   Need to look into what is going on.

**Comment by rtodling** [comment]
> While comparing x0048 and SWELL analysis increments generated by assimilating each satellite sensor, the GPSRO increments looked too different, with opposite signs, at least for the Ps increment. Need to look into what is going on.

@mjkagnes123 Really? Can you show a fig? That's definitely an instrument I have not looked at ...

**Comment by mjkagnes123** [comment]
![GPSRO](https://github.com/user-attachments/assets/bd1520ef-dacc-4b6c-8214-67f88df0486f)

**Comment by rtodling** [comment]
Thanks @mjkagnes123  I suggest you open an issue on the above so we can continue the discussion there and not clobber this Epic Issue.  I know I'm the one who asked for a fig ... so not complaining, just commenting.
